### PR TITLE
Add markdown extractor signal.

### PR DESCRIPTION
**Pull request by nsthorat**
Extracts markdown spans, and emits the language of the markdown block.

https://lilacai-nikhil-staging.hf.space/datasets#local/glaive&expandedStats=%7B%22question.markdown_extractor.*.language%22%3Atrue%7D&query=%7B%22filters%22%3A%5B%7B%22path%22%3A%5B%22question%22%2C%22markdown_extractor%22%2C%22*%22%2C%22language%22%5D%2C%22op%22%3A%22equals%22%2C%22value%22%3A%22python%22%7D%5D%7D&schemaCollapsed=false

image

### CODE REVIEW

File: lilac/signals/markdown_code_block.py [hunk 1]
```diff
@@ -0,0 +1,46 @@
+"""Finds markdown blocks.
```
Comment (dsmilkov) [inline lilac/signals/markdown_code_block.py:1]: code blocks

File: lilac/signals/markdown_code_block.py [hunk 2]
```diff
@@ -0,0 +1,46 @@
...
+"""
+import re
+from typing import ClassVar, Iterable, Iterator, Optional, cast
+
+from typing_extensions import override
+
+from ..schema import Field, Item, RichData, field, span
+from ..signal import TextSignal
+
+markdown_block_re = re.compile('```([^\n ]*?)\n(.*?)\n```', re.MULTILINE | re.DOTALL)
```
Comment (dsmilkov) [inline lilac/signals/markdown_code_block.py:14]: maybe this can be a class variable since these little things add up in import time (compilation of regex)

**Review by dsmilkov — APPROVED** [review]

File: lilac/signals/markdown_code_block.py (hunk 1 above)
Comment (nsthorat) [reply lilac/signals/markdown_code_block.py]: done

File: lilac/signals/markdown_code_block.py (hunk 2 above)
Comment (nsthorat) [reply lilac/signals/markdown_code_block.py]: done

**Pull request merged**